safe_transaction_service/history/services/balance_service.py

```diff
@@ -156,22 +158,49 @@ def get_balances(
             safe_address
         ).count()
         cache_key = (
-            f"balances:{safe_address}:{only_trusted}:{exclude_spam}:"
+            f"balances:{safe_address}:{only_trusted}:{exclude_spam}:{limit}:{offset}"
             f"{number_erc20_events}:{number_eth_events}:{events_sending_eth}"
         )
+        cache_key_count = f"balances:{safe_address}:{only_trusted}:{exclude_spam}"
         if balances := django_cache.get(cache_key):
-            return balances
+            count = django_cache.get(cache_key_count)
+            return balances, count
         else:
-            balances = self._get_balances(safe_address, only_trusted, exclude_spam)
+            balances, count = self._get_balances(
+                safe_address, only_trusted, exclude_spam, limit, offset
+            )
             django_cache.set(cache_key, balances, 60 * 10)  # 10 minutes cache
-            return balances
+            django_cache.set(cache_key_count, count, 60 * 10)  # 10 minutes cache
+            return balances, count
+
+    def _get_erc20_list_limits(self, limit: int, offset: int) -> Tuple[int, int]:
```

Comment:
Not sure if this functions is a little bit confusing, what do you think?

Reply:
Maybe better return just the list and pass the list as parameter?

Reply:
I don't have a strong opinion here, but indeed sounds confusing

Reply:
Done https://github.com/safe-global/safe-transaction-service/pull/2118/commits/0d1a9ab5800464f18a1dbb2287cd4ce9e8ad90f2